Firebase/Auth/Source/Auth/FIRAuth.m

```diff
@@ -439,7 +441,12 @@ - (nullable instancetype)initWithAPIKey:(NSString *)APIKey appName:(NSString *)a
           appCredentialManager:strongSelf->_appCredentialManager];
 
       [GULAppDelegateSwizzler registerAppDelegateInterceptor:strongSelf];
-      #endif
+      #if ((TARGET_OS_IOS || TARGET_OS_TV) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= 130000))
+        if (@available(iOS 13, tvos 13, *)) {
```

Comment:
nit: I think this part can be changed in the same way as `[GULSceneDelegateSwizzler proxyOriginalSceneDelegate];`. Doesn't look like a blocker to me though.

Reply:
The interceptor is a class of id<UISceneDelegate>, let me see if we can do the same.

Reply:
Seems we have to keep the checks because of `UISceneDelegate`.